skfp/applicability_domain/knn.py

```diff
@@ -0,0 +1,242 @@
+        Aggregation method for distances to k nearest neigbors:
+            - "mean": use the mean distance to k neighbors,
+            - "max": use the maximum distance among k neighbors,
+            - "min": use the distance to the closest neigbor.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See scikit-learn documentation on ``n_jobs`` for more details.
+
+    verbose : int or dict, default=0
+        Controls the verbosity when filtering molecules.
+        If a dictionary is passed, it is treated as kwargs for ``tqdm()``,
+        and can be used to control the progress bar.
+
+    References
+    ----------
+
+    .. [1] `Klingspohn, W., Mathea, M., ter Laak, A. et al.
+        "Efficiency of different measures for defining the applicability domain of
+        classification models."
+        J Cheminform 9, 44 (2017)
+        <https://doi.org/10.1186/s13321-017-0230-2>`_
+
+    .. [2] `Harmeling, S., Dornhege G., Tax D., Meinecke F., Müller KR.
+        "From outliers to prototypes: Ordering data."
+        Neurocomputing, 69, 13, pages 1608-1618, (2006)
+        <https://doi.org/10.1016/j.neucom.2005.05.015>`_
+
+    .. [3] `Kar S., Roy K., Leszczynski J.
+        "Applicability Domain: A Step Toward Confident Predictions and Decidability for QSAR Modeling"
+        Methods Mol Biol, 1800, pages 141-169, (2018)
+        <https://doi.org/10.1007/978-1-4939-7899-1_6>`_
+
+    Examples
+    --------
+    >>> from skfp.applicability_domain import KNNADChecker
+    >>> import numpy as np
+    >>> X_train_binary = np.array([
+    ...     [1, 1, 1],
+    ...     [0, 1, 1],
+    ...     [0, 0, 1]
+    ... ])
+    >>> X_test_binary = 1 - X_train_binary
+    >>> knn_ad_checker_binary = KNNADChecker(k=2, metric="tanimoto_binary_distance", agg="mean")
+    >>> knn_ad_checker_binary
+    KNNADChecker()
+
+    >>> knn_ad_checker_binary.fit(X_train_binary)
+    KNNADChecker()
+
+    >>> knn_ad_checker_binary.predict(X_test_binary)
+    array([False, False, False])
+
+    >>> X_train_count = np.array([
+    ...     [1.2, 2.3],
+    ...     [3.4, 4.5],
+    ...     [5.6, 6.7]
+    ... ])
+    >>> X_test_count = X_train_count + 10
+    >>> knn_ad_checker_count = KNNADChecker(k=2, metric="tanimoto_count_distance", agg="min")
+    >>> knn_ad_checker_count
+    KNNADChecker()
+
+    >>> knn_ad_checker_count.fit(X_train_count)
+    KNNADChecker()
+
+    >>> knn_ad_checker_count.predict(X_test_count)
+    array([False, False, False])
+
+    """
+
+    _parameter_constraints: dict = {
+        **BaseADChecker._parameter_constraints,
+        "k": [Interval(Integral, 1, None, closed="left")],
+        "metric": [
+            callable,
+            StrOptions(METRIC_NAMES),
```

Comment:
If you remove comma after `METRIC_NAMES)`, I think this will collapse nicely into one line